receiver/kubeletstatsreceiver/config.go

```diff
@@ -103,5 +117,19 @@ func (cfg *Config) Unmarshal(componentParser *confmap.Conf) error {
 		cfg.MetricGroupsToCollect = defaultMetricGroups
 	}
 
+	if cfg.NodeName == "" {
+		value, envExists := os.LookupEnv("K8S_NODE_NAME")
```

Comment:
I don't think we should support this. It doesn't seem like it matches the documentation that suggests explicitly setting the `node_name` from this env var. Do we have any instances in the collector codebase where we support the env vars like this OOTB? I don't like committing to support env variables that will be hard to change going forward in case we establish another naming schema as was proposed in https://github.com/open-telemetry/oteps/pull/195

Reply:
k8sattributes supports this feature, but I agree that it is an anti-pattern and we should rely entirely on the config

Reply:
Thank's for pointing this out. I have removed that.